test/parallel/xyk-rpc.error.test.ts

```diff
@@ -0,0 +1,66 @@
+import {getApi, initApi} from "../../utils/api";
+import { calculate_buy_price_rpc, calculate_sell_price_rpc} from '../../utils/tx'
+import BN from 'bn.js'
+
+
+jest.spyOn(console, 'log').mockImplementation(jest.fn());
+jest.setTimeout(1500000);
+process.env.NODE_ENV = 'test';
+
+beforeAll( async () => {
+	try {
+		getApi();
+	  } catch(e) {
+		await initApi();
+	}
+
+})
+
+
+test.each([
+		[new BN(-1),new BN(-1),new BN(-1),"createType(Balance):: Balance: Negative number passed to unsigned type"],
+		[new BN(1),new BN(-1),new BN(1),"createType(Balance):: Balance: Negative number passed to unsigned type"],
+		[new BN(-1),new BN(1),new BN(1),"createType(Balance):: Balance: Negative number passed to unsigned type"],
+	])
+	('xyk-rpc - calculate_sell_price validates parameters - Negative params', async(input_reserve,output_reserve,amount, expected) => {
+
+	await expect( 
+		calculate_sell_price_rpc(input_reserve,output_reserve, amount)
+	).rejects
+	.toThrow(expected.toString())
+
+	await expect( 
+		calculate_buy_price_rpc(input_reserve,output_reserve, amount)
+	).rejects
+	.toThrow(expected.toString())
+	
+});
+
+test.each([
+	[new BN(1),new BN(1),new BN(0),new BN(0)],
+	[new BN(0),new BN(0),new BN(0),new BN(0)],
+	[new BN(0),new BN(1),new BN(1),new BN(1)],
+	[new BN(0),new BN(0),new BN(1),new BN(0)],
```

Comment:
The pool can not exist in these states. So they are not semantically significant, but I guess you can use them to test if the rpc is behaving as expected (or some subtle behaviour has changed).

Reply:
that's the idea!